BUILD.md

```diff
@@ -318,20 +326,33 @@ DOCKER_TRACE=1 bin/docker-build-proxy
 ### Updating protobuf dependencies
  If you make Protobuf changes, run:
  ```bash
-bin/dep ensure
 bin/protoc-go.sh
 ```
 
 ### Updating Docker dependencies
 
-The go Docker images rely on base dependency images with
+The Go Docker images rely on base dependency images with
 hard-coded SHA's:
 
 `gcr.io/linkerd-io/go-deps` depends on
-- [`Gopkg.lock`](Gopkg.lock)
+- [`go.mod`](go.mod)
 - [`Dockerfile-go-deps`](Dockerfile-go-deps)
 
-`bin/update-go-deps-shas` must be run when go dependencies change.
+`bin/update-go-deps-shas` must be run when Go dependencies change.
+
+### Updating ServiceProfile generated code
+
+The [ServiceProfile client code](./controller/gen/client) is generated by
+[`bin/update-codegen.sh`](bin/update-codegen.sh), which depends on
+[K8s code-generator](https://github.com/kubernetes/code-generator), which does
+not yet support Go Modules. To re-generate this code, check out this repo into
```

Comment:
For an example of doing this with Go Modules, see https://github.com/deislabs/smi-sdk-go/blob/master/hack/update-codegen.sh

Reply:
filed #3145 to track, thanks!